src/main/java/io/vertx/core/spi/JsonFactory.java

```diff
@@ -21,7 +22,32 @@
  */
 public interface JsonFactory {
 
-  JsonFactory INSTANCE = Json.load();
+  /**
+   * <p> Load the JSON factory with the {@code ServiceLoader}
+   *
+   * <ul>
+   *   <li>An attempt is made to load a factory using the service loader {@code META-INF/services} {@link JsonFactory}.</li>
+   *   <li>Factories are sorted </li>
+   *   <li>If not factory is resolved (which is usually the default case), {@link JacksonFactory#INSTANCE} is used.</li>
+   * </ul>
+   *
+   * <p> When the default Jackson codec is used and {@code jackson-databind} is available then a codec using it
+   * will be used otherwise the codec will only use {@code jackson-core} and provide best effort mapping.
+   */
+  static JsonFactory load() {
+    return Utils.load();
+  }
+
+  /**
+   * The order of the factory. If there is more than one matching factory they will be tried in ascending order.
+   *
+   * @implSpec returns {@link Integer#MAX_VALUE}
+   *
+   * @return  the order
+   */
+  default int order() {
+    return Integer.MAX_VALUE;
```

Comment:
Shouldn't you use a lower value?

If an implement omits that method (because it does not care) it will be always used.

Reply:
Typically in your test, if one of the implementations does not provide the method, it will be first.

Reply:
actually it sorts by ascending order and takes the first one, so factory that don't implement this will always be chosen last.

Reply:
Oh sorry! Misread it! 

So all good for me.